### ISSUE TITLE
embed or link to graphiql that honors the headers?

### ISSUE THREAD

**Issue by maxandersen** [quarkusio/quarkus#33179]
### Description

would be nice to somehow have dev UI embed graphiql or link/provide info on how to run graphiql or similar to explore the setup graphql endpoints.

especially useful if it can setup headers so things like auth bearer will be honored (i.e. for https://api.github.com/graphql)

**Comment by phillip-kruger** [comment]
Are you talking about GraphQL Client ? W.r.t the server, if you have a GraphQL Service, that is secured, you can access the GraphiQL via the Security Dev UI Screen and it will pre-populate the authentication headers.

**Comment by maxandersen** [comment]
i have a graphql client that calls out. anyway to have it for that?

**Comment by phillip-kruger** [comment]
@maxandersen - if I follow you correctly, you want the graphiql for an external service ? Already secured ? 

@jmartisk w.d.y.t ?

**Comment by jmartisk** [comment]
You want graphiql to have an option to use a configured GraphQL client underneath (and thus call a remote service) instead of calling localhost?

**Comment by maxandersen** [comment]
Yeah. Imagine you are using graphql towards GitHub graphql. The client url is then known and should be available to do tests against. 

Wdyt?

**Comment by jmartisk** [comment]
We're currently working towards upgrading to GraphiQL 2.x, so after that is done, we can try and see if this would work. It might be useful.

Apart from the server-side extension providing a graphql-ui that runs against localhost, the client extension could provide a separate graphql-ui instance for each configured client.
